### ISSUE TITLE
Add runtime n-execution helpers

### ISSUE THREAD

**Issue by Mzack9999** [projectdiscovery/utils#186]
## Description
Many times it's necessary to execute once a function at package/struct level, with the requirement of declaring and leaving around many `sync.Once` field members. It would be useful to have a util package with the following functions signatures:
```go
// Execute the function once and associate it with a unique id
syncutil.DoOnce(id string, f func() {})

// Allow rexecuting the function
syncutil.Reset(id string)
```

Internally the reference can be kept with a sync-lock map

**Comment by tarunKoyalwar** [comment]
@Mzack9999 something similar to this https://pkg.go.dev/sync@master#pkg-index ??

**Comment by Mzack9999** [comment]
@tarunKoyalwar yeah, basically a static wrapper around `sync.Once` with a few additional methods like `.Done(..)` and `Reset(...)`

**Comment by Mzack9999** [comment]
Closing - Redundant with new `sync.OnceFunc(...)` and `sync.OnceValue(...)` from `go 1.21`
